b2g/chrome/content/remote_control.js

```diff
@@ -20,57 +20,113 @@ const kPrefPresentationDiscoverable = "dom.presentation.discoverable";
     }
   }
 
-  function setupRemoteControl() {
-    if (Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
-      importRemoteControlService();
-      remoteControlScope.RemoteControlService.start();
-    } else {
-      remoteControlScope.RemoteControlService.stop();
-    }
+  function getTime() {
+    return new Date().getTime();
+  }
+
+  function debug(s) {
+    let time = new Date(getTime()).toString();
+    // dump("[" + time + "] remote_control: " + s + "\n");
   }
 
-  Services.prefs.addObserver(kPrefPresentationDiscoverable, setupRemoteControl, false);
+  // Monitor the network status and the time change
+  var statusMonitor = {
+    init: function sm_init() {
+      debug("init");
+
+      Services.prefs.addObserver(kPrefPresentationDiscoverable, this, false);
+
+#ifdef MOZ_WIDGET_GONK
+      window.addEventListener('moztimechange', this.onTimeChange);
+
+      if (Ci.nsINetworkManager) {
+        Services.obs.addObserver(this, "network-active-changed", false);
+        Services.obs.addObserver(this, "network:offline-status-changed", false);
+      }
+#endif
```

Comment:
Hmm....the [nsINetworkManager.idl](https://dxr.mozilla.org/mozilla-central/source/dom/system/gonk/nsINetworkManager.idl) is under ```dom/system/gonk```[0]

Does partner's platform also has their own nsINetworkManager?

[0] https://dxr.mozilla.org/mozilla-central/source/dom/system/gonk/nsINetworkManager.idl